### PR TITLE
Optimize the closing logic of the note tooltip.

### PR DESCRIPTION
**Pull request by dvai**
When I move the mouse out of the tooltip, in most cases it means that I am no longer interested in it. Adding this change can avoid many unexpected behaviors, such as the tooltip still remaining on the interface after clicking a link in the tooltip.

### CODE REVIEW

File: src/public/app/services/note_tooltip.js [hunk 1]
```diff
@@ -93,6 +93,10 @@ async function mouseEnterHandler() {
             }
         }, 1000);
     }
+    // When the mouse is moved out, it means that the tooltip is no longer being attended to.
+    $(`.${linkId}`).on("mouseleave", () => {
+        setTimeout(() => $(this).tooltip('dispose'), 1000);
+    })
```
Comment (zadam) [inline src/public/app/services/note_tooltip.js]: How does this differ from the code a few lines above disposing of the tooltip after a second if mouse doesn't hover over it? Seems to be doing the same thing ...

**Comment by dvai** [comment]
@zadam  If this code segment is not added, the tooltip will not close automatically after my mouse hovers over it. You can compare these two videos:

before：

https://github.com/zadam/trilium/assets/33273040/efd4a02f-9204-41cf-ba15-f58322fdb0b2

after:


https://github.com/zadam/trilium/assets/33273040/832757ea-c8b0-40dd-8152-255ce2e67dfb

**Comment by rauenzi** [comment]
I tested both on stable and beta and I couldn't recreate any stuck tooltips. Can you provide an example the causes it consistently? I think it'd be good to understand why the existing code is insufficient (if it really is).

**Comment by zadam** [comment]
I managed to reproduce some of the stuck tooltips (on master) and I believe I fixed them.

I don't want to use the mouseleave event, because I want to allow the user to move the cursor over the tooltip to be able to e.g. scroll it and click links within it.

**Pull request closed without merging**